test/tests/docker-build.sh

```diff
@@ -16,9 +16,63 @@ set -e
 dir="$1"; shift
 [ -d "$dir" ]
 
+imageTag="$1"; shift
+image="${imageTag%%:*}"
+tag="${imageTag#*:}"
+[ "$image" != "$tag" ] || tag='latest'
+
 tmp="$(mktemp -t -d docker-library-test-build-XXXXXXXXXX)"
 trap "rm -rf '$tmp'" EXIT
 
 cat > "$tmp/Dockerfile"
+
+from="$(awk -F '[[:space:]]+' 'toupper($1) == "FROM" { print $2; exit }' "$tmp/Dockerfile")"
+onbuilds="$(docker inspect -f '{{len .Config.OnBuild}}' "$from")"
+if [ "$onbuilds" -gt 0 ]; then
+	# crap, the image we want to build has some ONBUILD instructions
+	# those are kind of going to ruin our day
+	# let's do some hacks to strip those bad boys out in a new fake layer
+	declare -A json=(
+		[Size]=0
+	)
+	declare -A mappings=(
+		[parent]='.Id'
+		[created]='.Created'
+		[container]='.Container'
+		[container_config]='.ContainerConfig'
+		[config]='.Config'
+		[docker_version]='.DockerVersion'
+		[architecture]='.Architecture'
+		[os]='.Os'
+	)
+	for key in "${!mappings[@]}"; do
+		val="$(docker inspect -f '{{json '"${mappings[$key]}"'}}' "$from")"
+		json["$key"]="$val"
+	done
+	onbuildConfig="$(docker inspect -f '{{json .Config.OnBuild}}' "$from" | sed 's/[]\/$*.^|[]/\\&/g')" # pre-escaped for use within "sed"
+	json[config]="$(echo "${json[config]}" | sed -r "s/$onbuildConfig/null/g")" # grab the image config, but scrub the onbuilds
+	jsonString='{'
+	first=1
+	for key in "${!json[@]}"; do
+		[ "$first" ] || jsonString+=','
+		first=
+		jsonString+='"'"$key"'":'"${json[$key]}"
+	done
+	newId="$(echo "$jsonString" | sha256sum | cut -d' ' -f1)" # lol, this is hacky
+	jsonString+=',"id":"'"$newId"'"}'
+	mkdir -p "$tmp/onbuild-hacks/$newId"
+	echo "$jsonString" > "$tmp/onbuild-hacks/$newId/json"
+	echo -n '1.0' > "$tmp/onbuild-hacks/$newId/VERSION"
+	dd if=/dev/zero of="$tmp/onbuild-hacks/$newId/layer.tar" bs=1k count=1 &> /dev/null # empty tar file
+	cat > "$tmp/onbuild-hacks/repositories" <<EOF
+{"$image":{"$tag":"$newId"}}
+EOF
+	tar -cC "$tmp/onbuild-hacks" . | docker load
+	rm -rf "$tmp/onbuild-hacks"
```

Comment:
I'm thinking that maybe this whole block deserves to be a whole new script that this script invokes directly instead...